contrib/asan.sh

```diff
@@ -0,0 +1,83 @@
+#!/bin/sh
+# This file is a part of Julia. License is MIT: https://julialang.org/license
+#
+# Usage:
+#     contrib/asan.sh <path> [<make_targets>...]
+#
+# Build ASAN-enabled julia checked out in the current directory.  Given a
+# workspace directory <path>, build ASAN-enabled julia in <path>/asan.  Required
+# toolss are install under <path>/toolchain.  This scripts also takes optional
+# <make_targets> arguments which are passed to `make`.  The default make target
+# is `debug`.
+
+set -ue
+
+# `$WORKSPACE` is a directory in which we careate `toolchain` and `asan`
+# sub-directories.
+WORKSPACE="$1"
+shift
+if [ "$WORKSPACE" = "" ]; then
+    echo "Workspace directory must be specified as the first argument" >&2
+    exit 2
+fi
+
+mkdir -pv "$WORKSPACE"
+WORKSPACE="$(cd "$WORKSPACE" && pwd)"
+if [ "$WORKSPACE" = "" ]; then
+    echo "Failed to create the workspace directory." >&2
+    exit 2
+fi
+
+echo
+echo "Installing toolchain..."
+
+TOOLCHAIN="$WORKSPACE/toolchain"
+if [ ! -d "$TOOLCHAIN" ]; then
+    make configure O=$TOOLCHAIN
```

Comment:
We should not assume that we're sitting in the top-level repository directory; we should have something like the following at the top:

```
JULIA_HOME="$(dirname "$( cd "$( dirname "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )" )"
```

And then we should `make -C "${JULIA_HOME}" configure O="${TOOLCHAIN}"` here.